lib/FlagDependencyExportsPlugin.js

```diff
@@ -9,21 +9,24 @@ class FlagDependencyExportsPlugin {
 	apply(compiler) {
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin("finish-modules", (modules) => {
-
 				const dependencies = Object.create(null);
 
 				let module;
 				let moduleWithExports;
+				let moduleProvidedExports;
 				const queue = modules.filter((m) => !m.providedExports);
 				for(let i = 0; i < queue.length; i++) {
 					module = queue[i];
 
 					if(module.providedExports !== true) {
 						moduleWithExports = false;
+						moduleProvidedExports = Array.isArray(module.providedExports) ? new Set(module.providedExports) : new Set();
```

Comment:
nope, as `module.providedExports` could be `true` and we would not want to have a `Set` with `true` as a field.